crowbar_framework/app/models/service_object.rb

```diff
@@ -989,20 +993,28 @@ def apply_role(role, inst, in_queue, bootstrap = false)
       end
     end
 
+    pre_cached_nodes = {}
+    filtered_elements = nil
+
     # When bootstrapping, we don't run chef, so there's no need for queuing
     if bootstrap
-      pre_cached_nodes = {}
       # do not try to process the queue in any case
       in_queue = true
     else
       # Attempt to queue the proposal.  If delay is empty, then run it.
       deps = proposal_dependencies(role)
+      filtered_elements = new_elements.deep_dup
+
+      if skip_unchanged_nodes_enabled
+        filtered_elements = skip_unchanged_nodes(new_elements, old_role, role)
```

Comment:
To be future-proof, s/new_elements/filtered_elements/ in case we add some filter before this one.

Reply:
ah thats a good one